### PR TITLE
chore(cli): updates scaffolding to use new template

### PR DESCRIPTION
**Pull request by idea404**
updates scaffolding to use git commands instead of forge init command and specifies commit hash of the template repository for compatibility

task: https://app.asana.com/0/1206208509925075/1207453993335053/f

### CODE REVIEW

File: cli/cmd/develop.go [hunk 1]
```diff
@@ -47,35 +51,55 @@ type developerForgeProjectConfig struct {
...
+	cmd := exec.CommandContext(ctx, "git", "clone", "https://github.com/omni-network/"+cfg.templateName+".git", destinationPath)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return errors.Wrap(err, "failed to clone repository")
+	}
+
+	// Check out the specific commit if using the default template
+	if cfg.templateName == defaultTemplate {
+		cmd = exec.CommandContext(ctx, "git", "checkout", "1d0ba3c") // Commit hash of the template, update as needed
```
Comment (kevinhalliday) [inline cli/cmd/develop.go:71]: Maybe move this & it's comment to a constant, so it's clear it's a thing

File: cli/cmd/develop.go [hunk 2]
```diff
@@ -47,35 +51,55 @@ type developerForgeProjectConfig struct {
...
+		if err := cmd.Run(); err != nil {
+			return errors.Wrap(err, "failed to checkout commit")
 		}
 	}
 
-	// Execute the `forge init` command.
-	// #nosec G204
-	cmd := exec.Command("forge", "init", "--template", "https://github.com/omni-network/"+cfg.templateName+".git")
+	// Initialize submodules
+	cmd = exec.CommandContext(ctx, "git", "submodule", "update", "--init", "--recursive")
```
Comment (kevinhalliday) [inline cli/cmd/develop.go:81]: `forge` install`? same thing basically. though with git you don't need forge installed which is nice

File: cli/cmd/develop.go [hunk 3]
```diff
@@ -47,35 +51,55 @@ type developerForgeProjectConfig struct {
...
-			Suggest: "You can install foundry by visiting https://github.com/foundry-rs/foundry",
+// newForgeProjectTemplate creates a new project by cloning a repository, checking out a specific commit if necessary, and initializing a new Git repository.
+func newForgeProjectTemplate(ctx context.Context, cfg developerForgeProjectConfig) error {
+	destinationPath, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "failed to get current working directory")
+	}
+
+	// Clone the repository
+	cmd := exec.CommandContext(ctx, "git", "clone", "https://github.com/omni-network/"+cfg.templateName+".git", destinationPath)
```
Comment (kevinhalliday) [inline cli/cmd/develop.go:62]: Since we are removing `.git` and need no history, do `--depth 1` in clone

**Review by pavelnikolov — APPROVED** [review]

File: cli/cmd/develop.go (hunk 1 above)
Comment (idea404) [reply cli/cmd/develop.go]: Good idea did this

File: cli/cmd/develop.go (hunk 2 above)
Comment (idea404) [reply cli/cmd/develop.go:83]: Kept the git, as we now can support scaffolding `omni-network` templates that don't use foundry

File: cli/cmd/develop.go (hunk 3 above)
Comment (idea404) [reply cli/cmd/develop.go]: good one, added

**Pull request merged**